runtime/nvqir/custatevec/CuStateVecState.h

```diff
@@ -36,19 +36,6 @@
   };
 
 namespace cudaq {
-template <typename T>
```

Comment:
Are the files in this directory meant to interface with custatevec primarily or with CUDA-Q primarily?

It seems to suggest a possible sharp edge? If they are meant to link with custatevec and CUDA code, then why would they need to be changed to be C++ code? But if they are meant to link into CUDA-Q, then it makes some sense to have them be pure C++ code. But it remains a question: do we have an issue here when some code is built with CUDA and some with CUDA-Q in so far as linking them together? Why?

Reply:
Indeed, some code in this directory (only the .cu file) is built with cuda, while the rest is build with the same compiler as CUDA-Q. No, we do not have a problem linking them together. In fact, this should now properly clean up any accidental incompatibilities because the boundary between them is now cleanly defined by the CuStateVecSimulator.h header file. 

Note that both directions of the inclusion work without issues, as long as the build is properly configured and the boundary between different libraries is mindful of defining a compiler/std lib agnostic API and header files: It is possible to use a library compiled with gcc from a CUDA-Q library and compile with CUDA-Q, and it is possible to use a CUDA-Q library from code compiled with gcc (see the docs I added). The first direction here is made easier by the fact that the cuquantum and cuda libraries we depend on only have a gcc_s dependencies and no explicit libstdc++ dependency.